pkg/apis/crd/v1beta1/types.go

```diff
@@ -151,3 +151,267 @@ type ControllerCondition struct {
+	TTL int32 `json:"ttl,omitempty" yaml:"ttl,omitempty"`
+	// Flags is the flags for IP.
+	Flags int32 `json:"flags,omitempty" yaml:"flags,omitempty"`
+}
+
+// IPv6Header describes spec of an IPv6 header.
+type IPv6Header struct {
+	// NextHeader is the IPv6 protocol.
+	NextHeader *int32 `json:"nextHeader,omitempty" yaml:"nextHeader,omitempty"`
+	// HopLimit is the IPv6 Hop Limit.
+	HopLimit int32 `json:"hopLimit,omitempty" yaml:"hopLimit,omitempty"`
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	ICMP *ICMPEchoRequestHeader `json:"icmp,omitempty" yaml:"icmp,omitempty"`
+	UDP  *UDPHeader             `json:"udp,omitempty" yaml:"udp,omitempty"`
+	TCP  *TCPHeader             `json:"tcp,omitempty" yaml:"tcp,omitempty"`
+}
+
+// ICMPEchoRequestHeader describes spec of an ICMP echo request header.
+type ICMPEchoRequestHeader struct {
+	// ID is the ICMPEchoRequestHeader ID.
+	ID int32 `json:"id,omitempty"`
+	// Sequence is the ICMPEchoRequestHeader sequence.
+	Sequence int32 `json:"sequence,omitempty"`
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	// SrcPort is the source port.
+	SrcPort int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort int32 `json:"dstPort,omitempty"`
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	// SrcPort is the source port.
+	SrcPort int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort int32 `json:"dstPort,omitempty"`
+	// Flags are flags in the header.
+	Flags *int32 `json:"flags,omitempty"`
+}
+
+// Packet includes header info.
+type Packet struct {
+	SrcIP string `json:"srcIP,omitempty"`
+	DstIP string `json:"dstIP,omitempty"`
+	// Length is the IP packet length (includes the IPv4 or IPv6 header length).
+	Length          uint16          `json:"length,omitempty"`
+	IPHeader        *IPHeader       `json:"ipHeader,omitempty"`
+	IPv6Header      *IPv6Header     `json:"ipv6Header,omitempty"`
+	TransportHeader TransportHeader `json:"transportHeader,omitempty"`
+}
+
+// TraceflowStatus describes current status of the traceflow.
+type TraceflowStatus struct {
+	// Phase is the Traceflow phase.
+	Phase TraceflowPhase `json:"phase,omitempty"`
+	// Reason is a message indicating the reason of the traceflow's current phase.
+	Reason string `json:"reason,omitempty"`
+	// StartTime is the time at which the Traceflow as started by the Antrea Controller.
+	// Before K8s v1.20, null values (field not set) are not pruned, and a CR where a
+	// metav1.Time field is not set would fail OpenAPI validation (type string). The
+	// recommendation seems to be to use a pointer instead, and the field will be omitted when
+	// serializing.
+	// See https://github.com/kubernetes/kubernetes/issues/86811
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+	// DataplaneTag is a tag to identify a traceflow session across Nodes.
+	DataplaneTag uint8 `json:"dataplaneTag,omitempty"`
```

Comment:
Even in this case, I think we can easily use a signed integer. This is entirely managed by Antrea.

Reply:
We uses lower 6 bits in this value as DSCP, thus using int8 can work.

Reply:
Done